.github/workflows/deploy_helm_chart.yaml

```diff
@@ -0,0 +1,95 @@
+name: Deploy Helm Chart to FNS GKE
+
+on:
+  workflow_dispatch:
+    inputs:
+      name:
+        description: "Manual trigger for Helm chart deploy"
+      version:
+        description: "Version of the Helm chart to deploy (optional, defaults to latest  snapshot)"
+        required: false
+      environment:
+        description: "Environment to deploy to"
+        required: true
+        default: "test"
+        options:
+          - "test"
+          - "prod"
+
+jobs:
+  setup-deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment }}
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+    steps:
+      - name: Checkout Fairspace repository
+        uses: actions/checkout@v4
+
+      - name: Checkout FNS model and values files
+        uses: actions/checkout@v4
+        with:
+          repository: 'thehyve/fnscloud-demonstrator'
+          path: './fns'
+          token: ${{ secrets.FNS_PAT }}
+
+      - name: Set environment
+        run: |
+          if [ -n "${{ github.event.inputs.environment }}" ]; then
+            ENVIRONMENT="${{ github.event.inputs.environment }}"
+          else
+            ENVIRONMENT="test"
+          fi
+          echo "ENVIRONMENT=$ENVIRONMENT" >> $GITHUB_ENV
+          echo "Starting deployment to the $ENVIRONMENT environment"
+
+      - name: Define Helm Chart version (taken from input or latest snapshot)
+        run: |
+          VERSION=$(cat ./VERSION)-SNAPSHOT
+          if [ -n "${{ github.event.inputs.version }}" ]; then
+            VERSION=${{ github.event.inputs.version }}
+          fi
+          echo "Helm charts version to be deployed: $VERSION"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Auth to Google Cloud
+        uses: google-github-actions/auth@v2
+        with:
+          workload_identity_provider: ${{ vars.CICD_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ vars.CICD_SERVICE_ACCOUNT_EMAIL }}
+
+      - name: Get GKE credentials to access the cluster
+        uses: google-github-actions/get-gke-credentials@v2
+        with:
+          cluster_name: fns-cloud-f01
+          location: europe-west1
+
+      - name: Setup Helm
+        uses: azure/setup-helm@v4.1.0
+        with:
+          version: ${{ vars.HELM_VERSION }}
+
+      - name: Login with Helm
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | helm registry login ${{ vars.HELM_REGISTRY }} --username ${{ github.repository_owner }} --password-stdin
+
+      - name: Delete StatefulSet (if exists) to force re-deploy
```

Comment:
I think it would be good to check if the namespace exists first and if not - create it

Reply:
good point, added a step to check if ns exists. It will create a new one if it doesn't. Also, checked that delete stateful-set step works as expected (not failing if ns does not exists)